forge-gui/res/cardsfolder/upcoming/jon_irenicus_shattered_one.txt

```diff
@@ -0,0 +1,15 @@
+Name:Jon Irenicus, Shattered One
+ManaCost:2 U B
+Types:Legendary Creature Elf Wizard
+PT:3/3
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ DBSelectRecipient | TriggerDescription$ At the beginning of your end step, target opponent gains control of up to one target creature you control.
+SVar:DBSelectRecipient:DB$ Pump | ValidTgts$ Opponent | TgtPrompt$ Select target opponent | SubAbility$ DBDonate
+SVar:DBDonate:DB$ GainControl | ValidTgts$ Creature.YouCtrl | TargetMin$ 0 | TargetMax$ 1 | TgtPrompt$ Select up to one target creature you control | NewController$ ParentTarget | SubAbility$ DBPutCounters
+SVar:DBPutCounters:DB$ PutCounter | Defined$ Targeted | CounterType$ P1P1 | CounterNum$ 2 | SubAbility$ DBTap
```

Comment:
This one probably needs to be TargetedCard so it doesn't affect the targeted player

Reply:
I tried replacing
```
SVar:DBPutCounters:DB$ PutCounter | Defined$ Targeted | CounterType$ P1P1 | CounterNum$ 2 | SubAbility$ DBTap
```
just now with
```
SVar:DBPutCounters:DB$ PutCounter | Defined$ TargetedCard | CounterType$ P1P1 | CounterNum$ 2 | SubAbility$ DBTap
```
and it doesn't seem to work. The +1/+1 counters are no longer added. Was that how I was supposed to amend it?

Reply:
@tool4ever ⬆️ can you debug what is funky there?

Reply:
So should the script be left as is and merged, or should AbilityUtils changes be made first?

Reply:
Seems like if Defined contains "Card" then we might skip some steps here?

![image](https://user-images.githubusercontent.com/103371817/177824745-b6e9cbb0-c0d2-4faa-863b-7ca231d274e2.png)

Reply:
So probably just leave it alone for now and merge this as is?

Reply:
yikes, yeah that's no good 😆 

Maybe for a band-aid, add `RememberControlled$ True` to the GainControl line and then PutCounter `Defined$ Remembered` ??

Then it would need cleanup too but might be OK short-term fix?

Reply:
Sure – I like that fix better

Reply:
My latest commit adds `TargetedCard` to `getDefinedCards()` @tool4ever suggested. I also added it to `getDefinedPlayers()` so that it is properly dealt with (no players should be returned). Without the addition to `getDefinedPlayers()`, that function was returning both players and counters were being added to both.